scl/paloalto/panos.conf

```diff
@@ -0,0 +1,97 @@
+        prefix("`prefix`")
+        delimiters(',')
+      );
+    };
+  } elif (match('TRAFFIC', value('.panos.type'))) {
+      parser {
+        csv-parser(
+          columns("future_use1","receive_time","serial_number","type","log_subtype","version","generated_time","src_ip","dest_ip","src_translated_ip","dest_translated_ip","rule","src_user","dest_user","app","vsys","src_zone","dest_zone","src_interface","dest_interface","log_forwarding_profile","future_use3","session_id","repeat_count","src_port","dest_port","src_translated_port","dest_translated_port","session_flags","transport","action","bytes","bytes_out","bytes_in","packets","start_time","duration","http_category","future_use4","sequence_number","action_flags","src_location","dest_location","future_use5","packets_out","packets_in","session_end_reason","devicegroup_level1","devicegroup_level2","devicegroup_level3","devicegroup_level4","vsys_name","dvc_name")
+        prefix("`prefix`")
+        delimiters(',')
+      );
+    };
+  } elif (match('HIPWATCH', value('.panos.type'))) {
+      parser {
+        csv-parser(
+        columns("future_use1","receive_time","serial_number","type","log_subtype","version","generated_time","src_user","vsys","host_name","os","src_ip","hip_name","hip_count","hip_type","future_use3","future_use4","sequence_number","action_flags","devicegroup_level1","devicegroup_level2","devicegroup_level3","devicegroup_level4","vsys_name","dvc_name")
+        prefix("`prefix`")
+        delimiters(',')
+      );
+    };
+  } elif (match('CORRELATION', value('.panos.type'))) {
+      parser {
+        csv-parser(
+        columns("future_use1","receive_time","serial_number","type","log_subtype","version","generated_time","src_ip","src_user","vsys","category","severity","devicegroup_level1","devicegroup_level2","devicegroup_level3","devicegroup_level4","vsys_name","dvc_name","vsys_id","object","object_id","evidence")
+        prefix("`prefix`")
+        delimiters(',')
+      );
+    };
+  } elif (match('USERID', value('.panos.type'))) {
+      parser {
+        csv-parser(
+        columns("future_use1","receive_time","serial_number","type","log_subtype","version","generated_time","vsys","src_ip","source_name","event_id","repeat_count","timeout_threshold","src_port","dest_port","source","source_type","sequence_number","action_flags","devicegroup_level1","devicegroup_level2","devicegroup_level3","devicegroup_level4","vsys_name","dvc_name")
+        prefix("`prefix`")
+        delimiters(',')
+      );
+    };
+  };
+  rewrite {
+    set("paloalto_panos", value("PROGRAM"));
+    set("syslog", value("SOURCE"));};
```

Comment:
Is it necessary to override the source? It is a nice macro that refers back to the config file.

Reply:
If I do not set source I get next i my output message:
`"#anon-source0"`

Reply:
That should not be a parameter either. That macro has a[ documented](https://www.syslog-ng.com/technical-documents/doc/syslog-ng-open-source-edition/3.25/administration-guide/64#TOPIC-1349537) behavior. It should not be used for different purpose.

What is your goal here? Maybe there is another way to achieve that. Do you want to refer to the originating message format? Or just the fact that this comes from a syslog-ng instance? What will it be used for? Is it okay to set it into a custom field, like `prefix.<a good a value name>`? What would be a good name? `source` seems misleading for me, because the true source is pan-os. But it can arrive from many other sources, for example from file.

Reply:
I guess I was using idea from
[Creating your first block for the syslog-ng configuration library (SCL)](https://www.syslog-ng.com/community/b/blog/posts/creating-your-first-block-for-the-syslog-ng-configuration-library-scl)
but that block was for source and this one is for parser. Nevertheless my use is specific with inline configuration so I will remove it in this submit just to avoid confusion.